doc/lua_api.txt

```diff
@@ -8930,8 +8943,307 @@ Used by `minetest.add_particlespawner`.
         -- If set to a valid number 1-6, specifies the tile from which the
         -- particle texture is picked.
         -- Otherwise, the default behavior is used. (currently: any random tile)
+
+        -- Legacy definition fields
+
+        minpos = {x=0, y=0, z=0},
+        maxpos = {x=0, y=0, z=0},
+        minvel = {x=0, y=0, z=0},
+        maxvel = {x=0, y=0, z=0},
+        minacc = {x=0, y=0, z=0},
+        maxacc = {x=0, y=0, z=0},
+        minexptime = 1,
+        maxexptime = 1,
+        minsize = 1,
+        maxsize = 1,
+        -- The particles' properties are random values between the min and max
+        -- values.
+        -- applies to: pos, velocity, acceleration, expirationtime, size
+        -- If `node` is set, min and maxsize can be set to 0 to spawn
+        -- randomly-sized particles (just like actual node dig particles).
+    }
+
+### Modern definition fields
+
+After v5.6.0, spawner properties can be defined in several different ways depending
+on the level of control you need. `pos` for instance can be set as a single vector,
+in which case all particles will appear at that exact point throughout the lifetime
+of the spawner. Alternately, it can be specified as a min-max pair, specifying a
+cubic range the particles can appear randomly within. Finally, some properties can
+be animated by suffixing their key with `_tween` (e.g. `pos_tween`) and supplying
+a tween table.
+
+The following definitions are all equivalent, listed in order of precedence from
+lowest (the legacy syntax) to highest (tween tables). If multiple forms of a
+property definition are present, the highest-precidence form will be selected
+and all lower-precedence fields will be ignored, allowing for graceful
+degradation in older clients).
+
+    {
```

Comment:
@velartrill 

I overlooked these definitions because the syntax is the same as for object definitions in other sections, while you describe different data types accepted by the property.

I propose making the data types more prominent and make the following changes:
* Use `vector` instead of `vec3` because this is the only section that introduces `vec3` name.
* Declare `range` data type which can be a `vector range` and a `float range`.

For example, the section could look like this:

----

The following definitions are all equivalent, listed in order of precedence from
lowest (the legacy syntax) to highest (tween tables). If multiple forms of a
property definition are present, the highest-precidence form will be selected
and all lower-precedence fields will be ignored, allowing for graceful
degradation in older clients).

1. Old Syntax:

```
      maxpos = {x = 0, y = 0, z = 0},
      minpos = {x = 0, y = 0, z = 0},
```

2. Scalar value:

```
      pos = 0,
```
All components of every particle's position vector will be set to this value.

3. Vector:

```
      pos = vector.new(0,0,0),
```

All particles will appear at this exact position throughout the lifetime of the particlespawner.

5. Range

Ranges can be scalar (float) ranges and vector ranges. The following example shows a vector range:

```
      pos = {
            -- the particle will appear at a position that is picked at random from
            -- within a cubic range

            min = vector.new(0,0,0),
            -- `min` is the minimum value this property will be set to in particles
            -- spawned by the generator

            max = vector.new(0,0,0),
            -- `max` is the minimum value this property will be set to in particles
            -- spawned by the generator

            bias = 0,
            -- when `bias` is 0, all random values are exactly as likely as any
            -- other. when it is positive, the higher it is, the more likely values
            -- will appear towards the minimum end of the allowed spectrum. when
            -- it is negative, the lower it is, the more likely values will appear
            -- towards the maximum end of the allowed spectrum. the curve is
            -- exponential and there is no particular maximum or minimum value
        },
```

6.  Tween table:

A tween table should consist of a list of frames in the same form as the
untweened pos property above, which the engine will interpolate between,
and optionally a number of properties that control how the interpolation
takes place. currently **only two frames**, the first and the last, are
used, but extra frames are accepted for the sake of forward compatibility.
Any of the above definition styles can be used here as well in any combination
supported by the property type.

You need to append `_tween` suffix to the property name to assign a tween table.

```
        pos_tween = {
            style = "fwd",
            -- linear animation from first to last frame (default)
            style = "rev",
            -- linear animation from last to first frame
            style = "pulse",
            -- linear animation from first to last then back to first again
            style = "flicker",
            -- like "pulse", but slightly randomized to add a bit of stutter

            reps = 1,
            -- number of times the animation is played over the particle's lifespan

            start = 0.0,
            -- point in the spawner's lifespan at which the animation begins. 0 is
            -- the very beginning, 1 is the very end

            -- frames can be defined in a number of different ways, depending on the
            -- underlying type of the property. for now, all but the first and last
            -- frame are ignored

            -- frames

                -- floats
                0, 0,

                -- vec3s
                vector.new(0,0,0),
                vector.new(0,0,0),

                -- vec3 ranges
                { min = vector.new(0,0,0), max = vector.new(0,0,0), bias = 0 },
                { min = vector.new(0,0,0), max = vector.new(0,0,0), bias = 0 },

                -- mixed
                0, { min = vector.new(0,0,0), max = vector.new(0,0,0), bias = 0 },
        }
```